### ISSUE TITLE
[Bug]: dependency's top-level-await not executed in parallel

### ISSUE THREAD

**Issue by hi-ogawa** [rolldown/rolldown#3132]
### Reproduction link or steps

reproduction: https://stackblitz.com/edit/github-13kqqtad?file=dist%2Fentry.js

### input

- entry.js

```js
import "./dep1.js"
import "./dep2.js"
```

- dep1.js

```js
console.log("[dep1] sleep 1sec")
await new Promise(r => setTimeout(r, 1000))
console.log("[dep1] done")
```

- dep2.js

```js
console.log("[dep2] sleep 1sec")
await new Promise(r => setTimeout(r, 1000))
console.log("[dep2] done")
```

### output

```js
console.log("[dep1] sleep 1sec");
await new Promise(r => setTimeout(r, 1000));
console.log("[dep1] done");

console.log("[dep2] sleep 1sec");
await new Promise(r => setTimeout(r, 1000));
console.log("[dep2] done");
```

### What is expected?

I should work like an original esm:

```sh
❯ node src/entry.js
[dep1] sleep 1sec
[dep2] sleep 1sec
[dep1] done
[dep2] done
```

### What is actually happening?

```sh
❯ node dist/entry.js
[dep1] sleep 1sec
[dep1] done
[dep2] sleep 1sec
[dep2] done
```

### System Info

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    rolldown: 0.15.0-snapshot-3cea4f5-20241211003613 => 0.15.0-snapshot-3cea4f5-20241211003613
```

### Any additional comments?

I'm testing things after reading https://github.com/orgs/web-infra-dev/discussions/9 
As mentioned in the post, rollup and esbuild have the same behavior. I would imagine this is an out of scope for now, but I thought it's good to have an issue for other users to see.

I realized Vite SSR (module runner) also behaves similarly. I added examples in https://github.com/hi-ogawa/reproductions/tree/main/rolldown-top-level-await

**Comment by hyf0** [comment]
Yeah. Rolldown's lib output gonna follows what esbuild/rollup did. For app format, a custom module runtime allows to simulate TLA with correct semantic.

**Comment by hyf0** [comment]
> Yeah. Rolldown's lib output gonna follows what esbuild/rollup did. For app format, a custom module runtime allows to simulate TLA with correct semantic.

When the TLA is supported, the behavior would said above.

Supporting TLA is tracked at https://github.com/rolldown/rolldown/issues/3131, so I close this one.
